packages/react/src/components/MenuButton/MenuButton.tsx

```diff
@@ -0,0 +1,305 @@
+      {
+        open: true,
+        autoFocus: false, // focused by renderCallback
+      },
+      renderCallback,
+    )
+    e.preventDefault()
+  }
+
+  handleOpenChange = (e, { open }) => {
+    _.invoke(this.props, 'onOpenChange', e, { ...this.props, ...{ open } })
+    this.setState(() => ({
+      open,
+      autoFocus: true,
+    }))
+  }
+
+  handleMenuItemClick = (predefinedProps?: MenuItemProps) => (
+    e: React.SyntheticEvent,
+    itemProps: MenuItemProps,
+  ) => {
+    _.invoke(predefinedProps, 'onClick', e, itemProps)
+    if (!predefinedProps || !predefinedProps.menu) {
+      // do not close if clicked on item with submenu
+      this.setState({ open: false, autoFocus: false })
+    }
+  }
+
+  handleMenuItemOverrides = (menuItemAccessibilityAttributes: any) =>
+    _.map(_.get(this.props.menu, 'items'), (item: ShorthandValue<MenuItemProps>) =>
+      typeof item === 'object'
+        ? {
+            ...item,
+            onClick: this.handleMenuItemClick(item as MenuItemProps),
+            ...menuItemAccessibilityAttributes,
+          }
+        : {
+            content: item,
+            key: item,
+            onClick: this.handleMenuItemClick(),
+            ...menuItemAccessibilityAttributes,
+          },
+    )
+
+  renderComponent({
+    ElementType,
+    classes,
+    unhandledProps,
+    accessibility,
+    styles,
+  }: RenderResultConfig<ContextMenuProps>): React.ReactNode {
+    const { contextMenu, menu, ...popupProps } = this.props
+    const content =
+      menu &&
+      Menu.create(menu, {
+        defaultProps: {
+          ...accessibility.attributes.menu,
+          vertical: true,
+        },
+        overrideProps: {
+          items: this.handleMenuItemOverrides(accessibility.attributes.menuItem),
+        },
+      })
+
+    const overrideProps = {
+      open: this.state.open,
+      onOpenChange: this.handleOpenChange,
+      content: {
+        styles: styles.popupContent,
+        content: content && <Ref innerRef={this.menuRef}>{content}</Ref>,
+      },
+      children: undefined, // force-reset `children` defined for `Popup` as it collides with the `trigger
+    }
+
+    if (contextMenu) {
+      return Popup.create(popupProps, {
+        defaultProps: {
+          on: 'context',
+          trapFocus: true,
+          unstable_pinned: true,
+          shouldTriggerBeTabbable: false,
+        },
+        overrideProps,
+      })
+    }
+
+    return (
+      <ElementType
+        className={classes.root}
+        {...accessibility.attributes.root}
+        {...unhandledProps}
+        {...applyAccessibilityKeyHandlers(accessibility.keyHandlers.root, unhandledProps)}
+      >
+        <Ref innerRef={this.triggerRef}>
+          {Popup.create(popupProps, {
+            defaultProps: {
+              inline: true,
+              autoFocus: true,
+              unstable_pinned: true,
+            },
+            overrideProps: {
+              accessibility: () => accessibility,
```

Comment:
```suggestion
              accessibility,
```

Reply:
it's because of `TypeError: accessibility is not a function`. I'm not really happy about this construct, but I did not find any other way how to pass the pre-populated behavior to the Popup

Reply:
Covered offline, no other way now